SSE for Matrix4's Mult

32BitFelix opened the pull request:
Added an SSE(1) alternative to the Matrix4 structure's static Mult method

### Purpose of this PR

* Adding SIMD alternatives to the existing methods in Mathematics

### Testing status

* Trust me bro

### Comments

* Simply added a Mult method for Matrix4, utilizing SSE(1)

32BitFelix committed:
SSE for Matrix4's Mult

Added an SSE(1) alternative to the Matrix4 structure's static Mult method

NogginBops commented:
You need to add the appropriate preprocessor guard when using `System.Runtime.Intrinsics` for the build to succeed.
See how it's done here:
https://github.com/opentk/opentk/blob/master/src/OpenTK.Mathematics/Matrix/Matrix4.cs#L1681-L1697

32BitFelix committed:
Updated

Updated the implementation based on noggin's and boybaykiller's feedback. It still isn't as performant. Will avoid reference fixing for the next version

NogginBops commented:
Closing this in favor of #1899
